lib/guardian/topic_guardian.rb

```diff
@@ -48,8 +48,11 @@ def can_publish_topic?(topic, category)
   def can_create_topic?(parent)
     is_staff? ||
       (
-        user && user.trust_level >= SiteSetting.min_trust_to_create_topic.to_i &&
-          can_create_post?(parent) && Category.topic_create_allowed(self).limit(1).count == 1
+        user &&
+          (
+            user.trust_level >= SiteSetting.min_trust_to_create_topic.to_i ||
+              user.in_any_groups?(SiteSetting.create_topic_allowed_groups_map)
+          ) && can_create_post?(parent) && Category.topic_create_allowed(self).limit(1).count == 1
```

Comment:
Not related to your changes, but maybe we can opportunistically change this to be a bit more readable:

```suggestion
          ) && can_create_post?(parent) && Category.topic_create_allowed(self).any?
```

It creates a similar but simpler query, and it's easier to understand what it does. 🙂 

I let you decide whether to include or not.